ci-scripts/nightly_upload.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+set -euxo pipefail
+# nightly_upload: checks remote file for changes and uploads new version if necessary
+# Usage:
+# set scp(ssh) server with user and path
+# create env-variable with private key:
+# $ssh_server_key(delete first and last line)
+# $ssh_port e.g:3842
+# $ssh_address e.g.: user@www.internet.com:/root/ci/
+# call script: ./nightly_upload.sh -filename
+
+repo_name=$(basename `git rev-parse --show-toplevel`)
+echo repo_name
+ssh=$ssh_address$repo_name'/'
+port=$ssh_port
+build_file="$1"
+new_sha=$build_file"-sha256sum.txt"
+
+old_sha="old"-$new_sha
+date_now=$(date "+%F-%H-%M-%S")
+
+# create ssh-key file
+echo "-----BEGIN OPENSSH PRIVATE KEY-----" > ~/server_key
+echo "$ssh_server_key" >> ~/server_key
+echo "-----END OPENSSH PRIVATE KEY-----" >> ~/server_key
+
+chmod 600 ~/server_key
+cat ~/server_key
+
+sha256sum $build_file > $new_sha
+rsync -e "ssh -o StrictHostKeyChecking=accept-new -p $port -i ~/server_key" $ssh$new_sha $old_sha #TODO edge case there is fo file in folder!
+if cmp -s $new_sha $old_sha; then
+    echo Build is unchanged
+else
+    rsync -e "ssh -o StrictHostKeyChecking=accept-new -p $port -i ~/server_key" $new_sha $ssh$new_sha
```

Comment:
Can we replace StrictHostKeyChecking=accept-new with populating known_hosts file instead?
(moved from commit comment)

Reply:
I see. If we are going this way, then we should note this down in the documentation under potential risks.

Reply:
From my understanding, the only potential risk is that someone could trick us into uploading the nightly build to another server,  or is there also another risk factor?

Reply:
Ok, what I missed is the obvious danger of a MITM attack, which allows to modify the files sent to the server.
I am not sure if the secrets of the working instance are also in danger.

Reply:
added known_hosts file to docker img